### DISCUSSION TITLE
AMIP-type FHIST runs with L58 SE grid, cam_dev physics, updated CLUBB

### DISCUSSION THREAD

**Discussion by bstephens82** [NCAR/amwg_dev#93] [TF1: Unified physics for L58/L93]
I ran two 10-year simulations, one with `clubb_l_predict_upwp_vpwp` true and one false.  The runscript I used for the `clubb_l_predict_upwp_vpwp=.true.` test is [here](https://github.com/NCAR/amwg_dev/files/8334907/cam.runscript.simple.pmfT.colinparams.10yr.txt), although note that I did not set up the "resubmit" option properly (noob mistake) so I carried out my restarts manually.  The `clubb_l_predict_upwp_vpwp=.false.` runscript is identical apart from that flag setting.

Note that the version of CLUBB in the code base I used here is from November, not the more recent version that I think Kate is putting into the model, but I think the differences are likely to be minor.

The output from these runs is currently at /glade/scratch/stepheba/archive/pmfT.colin.10yr.b/ (prognostic=true) and /glade/scratch/stepheba/archive/pmfF.colin.10yr.b/ (prognostic=false).  I've also run diagnostics on these results and that output can be found at /glade/scratch/stepheba/cesm-postprocess/.

I'm not very familiar with all of the avaiable diagnostics yet, but here are some diagnostics comparing the two tests for sea-level pressure and surface stress.
![set3_ANN_PSL_c](https://user-images.githubusercontent.com/59519056/159754120-96af73c8-4c34-4c2e-862a-a6eb5d0c6042.png)
![set3_ANN_TAUX_OCEAN_c](https://user-images.githubusercontent.com/59519056/159754470-bfb86684-7f75-4eef-b572-349b2e43214d.png)

**Comment by adamrher** [comment]
interesting reductions in surface stress. Can you post the diagnostics on the web? To publish to http://webext.cgd.ucar.edu/ , go to /project/diagnostics/external/FHIST/ on any CGD machine, and follow examples to create directory to dump diagnostics into. If you don't have write permissions, email help@cgd.ucar.edu

**Comment by bstephens82** [comment]
I've gotten access to the directories Adam mentioned and posted the diagnostics online here: https://webext.cgd.ucar.edu/FHIST/pmfT.colin.10yr.b/.

**Comment by tto061** [comment]
[Pumas-ice-substep_tht_pages.pdf](https://github.com/NCAR/amwg_dev/files/8398565/Pumas-ice-substep_tht_pages.pdf)
Hi Adam
the weakning winds with prognostic momentum fluxes (and the concomitant, 
necessary re-ordering) is a result  have also got consistently in all my 
simulations (with FV L32). Will include some graphs here later.
Cheers
Thomas

Thomas Toniazzo
Meteorologiska institutionen (MISU)
Arrhenius Laboratory, Svante Arrhenius Väg 16C
Stockholms universitet     SE-106 91 Stockholm
Sverige                  tel. +46 (0)8 16 2416

**Comment by adamrher** [comment]
@bstephens82 @tto061 would you mind explaining to me what is the deal with this namelist ```clubb_l_diag_Lscale_from_tau```? My understanding is that this namelist controls whether to use all those tau variables. Are either of you using these?

I gleaned from @zarzycki that these tau variables are an "alternative" way of specifying the inverse time-scales used in various terms in the prognostic moments. However, I never knew the inverse time-scales to be so complicated. Are these tau variables then an enhancement to the default inverse time-scales, rather than just an alternative? How are the inverse time-scales specified in the default?

**Reply by bstephens82** [reply]
@adamrher I think the l_diag_Lscale_from_tau code was added shortly before my time working on CLUBB, and I think I've always used it, however I haven't worked on that code much myself.  From looking back at the history a bit, it looks like calculating Lscale directly (and then calculating the taus based on Lscale) was expensive and gave weak winds in hurricanes (this was back in 2018 so a lot may have changed this then).  So the decision was made to calculate the taus first (based on BV frequency and various tunable parameters) and then calculate Lscale based on that.  I believe all the various tau variables are used in either case.  If you can see this, the section of code that calculates Lscale first and then the taus is [here](https://github.com/larson-group/clubb_release/blob/a09324969523cc3c22a6ea6e203ff8ed970cd225/src/CLUBB_core/advance_clubb_core_module.F90#L1261-L1307), while the code that calculates the taus first and then Lscale is just below that section.

**Reply by zarzycki** [reply]
@adamrher The turbulent length scale can be currently diagnosed in CLUBB in two different ways. The default way (i.e., l_diag_Lscale_from_tau=.false.) is following Section 3b of [Golaz et al., 2002](https://journals.ametsoc.org/view/journals/atsc/59/24/1520-0469_2002_059_3540_apbmfb_2.0.co_2.xml). 

The newer way (l_diag_Lscale_from_tau=.true.) we have been applying in the CPT version(s) of CLUBB has been calculating Lscale as:

![Screen Shot 2022-03-25 at 12 34 35 PM](https://user-images.githubusercontent.com/10341827/160162732-0ea38ae5-fd58-4b7e-ba36-5e1d7e3fb227.png)

See Section 2b of [Nardi et al., 2022](https://journals.ametsoc.org/view/journals/mwre/aop/MWR-D-21-0186.1/MWR-D-21-0186.1.xml).

**Reply by adamrher** [reply]
I have to say, looking over the way Golaz et al. 2002 calculates L is a bit nauseating. It's like an entraining plume model using a fixed entrainment rate! The ```clubb_l_diag_Lscale_from_tau``` method seems like a more general approach. I take comfort in that.

**Comment by adamrher** [comment]
@bstephens82 @JulioTBacmeister I think it would be useful to do a run with prognostic momentum off, but with the ```clubb_l_diag_Lscale_from_tau``` still set to true. This is a thing, right? We are changing two big things at once in the run in this thread?

**Reply by bstephens82** [reply]
@adamrher the only change between the two runs linked in this thread is the value of clubb_l_predict_upwp_vpwp.  They are both run with l_diag_Lscale_from_tau=.true.

**Comment by bstephens82** [comment]
@adamrher I ran a 10-year FHIST SE experiment with clubb_l_diag_Lscale_from_tau=.true. and clubb_l_predict_upwp_vpwp=.false. starting from a clean 053 tag, and the results compared to obs are [here](https://webext.cgd.ucar.edu/FHIST/pmfF.LscaleflagT.old053tag/atm/pmfF.LscaleflagT.old053tag-obs.1980_1987/).  

I haven't looked closely at all the output yet, but since I've been looking at SWCF in my last few runs, I noticed this one has as hugely amplified SWCF:
![set5_ANN_SWCF_CERES-EBAF_obsc](https://user-images.githubusercontent.com/59519056/161112086-6bb946a4-a4dd-424b-966c-9055cbeef7bf.png)

**Reply by zarzycki** [reply]
A couple quick questions/thoughts for @bstephens82.

1.) What are the c_invrs_tau terms? I don't think we've taken any care to set any real defaults at this point.
2.) Have you "reset" the other coefficients (e.g, C1, etc.) that Zhun and Vince removed from the taus tuning deck?
3.) We've also been running `clubb_l_e3sm_config=.true.` -- is this .true. in your runs?

FWIW, this (getting way too much low cloud) was a common behavior when flipping on the new Lscale code without modifying clubb_gamma, etc. A deck of parameters I (extremely tentatively) optimized for L32 is:

```
clubb_C1          = 1.000000 ! Low Skewness in C1 Skewness Function. Units [-]
clubb_C1b         = 1.000000 ! High Skewness in C1 Skewness Function. Units [-]
!clubb_C1c         = 1.000000 ! Degree of Slope of C1 Skewness Function. Units [-]
clubb_C2rt        = 1.000000 ! C2 constant associated with rtp2 dissipation term. Units [-]
clubb_C2thl       = 1.000000 ! C2 constant associated with thlp2 dissipation term. Units [-]
clubb_C2rtthl     = 1.000000 ! C2 constant associated with rtpthlp dissipation term. Units [-]
clubb_C6rt        = 2.000000 ! Low Skewness in C6rt Skewness Function. Units [-]
clubb_C6rtb       = 2.000000 ! High Skewness in C6rt Skewness Function. Units [-]
clubb_C6rtc       = 1.000000 ! Degree of Slope of C6rt Skewness Function. Units [-]
clubb_C6thl       = 2.000000 ! Low Skewness in C6thl Skewness Function. Units [-]
clubb_C6thlb      = 2.000000 ! High Skewness in C6thl Skewness Function. Units [-]
clubb_C6thlc      = 1.000000 ! Degree of Slope of C6thl Skewness Function. Units [-]
clubb_C14         = 1.000000 ! Constant applied to up2 and vp2. Units [-]

clubb_l_diag_lscale_from_tau=.true.
clubb_l_e3sm_config=.true.

clubb_C_invrs_tau_bkgnd=3.727123755772682
clubb_C_invrs_tau_sfc=0.12743072568015346
clubb_C_invrs_tau_shear=0.12502726304767026
clubb_C_invrs_tau_N2=0.08122667220596895
clubb_c_uu_shr=0.1076484659222455
clubb_gamma_coef=0.5492223674353673
clubb_gamma_coefb=0.2531868210746816
clubb_c11=0.31057411754034614
clubb_c11b=0.3250718127387944
clubb_c8=3.440377776099962
clubb_beta=2.27756371212011
```

otherwise all namelist settings are default as of CESM2.2.

**Reply by bstephens82** [reply]
@zarzycki just for context Adam had asked if I could run a similar test starting from a clean 053 tag, so we could think about the effect of clubb_l_diag_Lscale_from_tau without the additional modification of using prognostic momentum fluxes (maybe this thread wasn't the best place for it in retrospect).  For the previous runs that I linked in this post, I was using all of your flags and parameters, just with a 58-level grid and turning on and off the prognostic momentum flux.  For this latest run, I started with the 053 tag and only modified the value of clubb_l_diag_Lscale_from_tau which is false by default in that tag.  For all other flags or parameters the model is using the default values in namelist_defaults_cam.xml or in the CLUBB code (the tau values are set in parameters_tunable.F90):
~~~
    C_invrs_tau_bkgnd         = 1.0_core_rknd,  &   !
    C_invrs_tau_sfc           = 0.1_core_rknd,  &   !
    C_invrs_tau_shear         = 0.02_core_rknd, &   !
    C_invrs_tau_N2            = 0.1_core_rknd,  &   !
    C_invrs_tau_N2_wp2        = 0.2_core_rknd,  &   !
    C_invrs_tau_N2_xp2        = 0.2_core_rknd,  &   !
    C_invrs_tau_N2_wpxp       = 0.0_core_rknd,  &   !
    C_invrs_tau_N2_clear_wp3  = 0.0_core_rknd
~~~
Same for the other CLUBB parameters--they are either being set in the namelist files or by the CLUBB code.

That is an interesting point about the SWCF response to the Lscale flag, I think @adamrher will be interested to hear that, I think this is kind of information he may have been interested in, although I was not aware that had been previously looked at.

**Reply by zarzycki** [reply]
>  For this latest run, I started with the 053 tag and only modified the value of clubb_l_diag_Lscale_from_tau which is false by default in that tag.

Ah, OK, that explains the SWCF. Essentially, clubb_l_diag_Lscale_from_tau just branches how the code diagnoses Lscale and is quite sensitive to CLUBB tuning settings. So in my opinion, while this is a CLUBB boolean, it behaves a lot less like other booleans since it leans hard on both the if/else in the code *and* those branches then lean hard on tuning coefficients (e.g., C14) that should set differently depending on the value of said boolean...

**Reply by adamrher** [reply]
I have some thoughts, but a quick question. One of the runs at the top of this thread w/ the new externals, has ```clubb_l_predict_upwp_vpwp=F``` and ```l_diag_Lscale_from_tau=T``` ... I believe it's called pmfF.colin.10yr.b. It's SWCF looks very reasonable and so all the clubb parameters such as (clubb_gamma) are being manually set to the values optimized by Colin, right? You're not using the default clubb namelists?

Whereas this latest run w/ the old externals is using the default namelists, and ```clubb_l_predict_upwp_vpwp=T```.

**Reply by zarzycki** [reply]
Yes, Adam, I believe that is correct.

The top run (`pmfF.colin.10yr.b`) includes all the settings I listed above in user_nl_cam. These have been "retuned" (again, extremely crudely, I need to be careful calling it an actual retune!) to behave well with the new CLUBB @Katetc is bringing in. Both of which use `clubb_l_diag_Lscale_from_tau=.true.` (since my retuning is contingent on having that set to true).

The simulation in this thread doesn't include the "retuned" coefficients, so default across the board, except Ben (@bstephens82, correct me if I'm wrong) only toggled `clubb_l_diag_Lscale_from_tau=.true.`, which activates the Lscale/tau code but uses the defaults from inside CLUBB/namelist_defaults -- these defaults do not behave overly well with the taus code ergo the SWCF goes through the roof.

**Reply by adamrher** [reply]
If I'm interpreting Coln's earlier post correctly (likely I'm not), I think we should have conditionals in the namelist_defaults that recovers our current namelists if ```l_diag_Lscale_from_tau=F``` and otherwise will use another complete set of clubb namelists if ```l_diag_Lscale_from_tau=T```. As a start we can plug in Colin's values above. But at least this will give something approaching backward compatibility if we run with ldiag=F. 

Does everyone agree? If so, I'll add this feedback to Kate's PR.

**Reply by bstephens82** [reply]
@adamrher I think you're interpreting correctly in the sense that the diag_Lscale flag has a big impact on the model and thus it might be reasonable to have different sets of CLUBB parameters depending on its value.  My only thought is that there are a lot of flags in CLUBB and one could imagine there being different sets of parameters that would be optimal for various flag settings, so it could conceivably get complicated, but if the CLUBB flag settings used in CAM are rarely changed (and I'm hazarding a guess that most CESM users don't mess with them?), then I think it could make sense.

**Reply by zarzycki** [reply]
@adamrher What you propose is probably the best path forward -- a conditional in namelist_defaults that has one set of coefficients for each of the `l_diag_Lscale_from_tau` options ... if `.false.` keep the default CLUBB settings, if `.true.` use updated tunings listed above (will certainly be updated, but for now let's just get them into the PR).

**Reply by adamrher** [reply]
@bstephens82 I think most users don't understand clubb well enough to start changing c6 or whatever. If anything, I'd say clubb_gamma is the most mainstream clubb namelist. But this ldiag flag is really a different ball park IMO. It really seems to alter the dissipative characteristics such that a wholesale retune of the main clubb knobs is required (that point being made by your SWCF plot!). I will ask Kate to do this.

**Reply by adamrher** [reply]
> 3.) We've also been running clubb_l_e3sm_config=.true. -- is this .true. in your runs?

This is true in Ben's runs at the top of this thread ... what does it do?

**Reply by bstephens82** [reply]
l_e3sm_config was introduced for backward-compatibility with the E3SM host model after some changes were made to CLUBB.  It affects the values of a few of the "taus", although I would have to do some digging to understand the significance of the changes (I actually introduced the flag but not the changes to the taus code).  My understanding at the time was that the flag also made the code backward-compatible with CAM.  The code can be found in CLUBB_core/mixing_length.F90:
~~~
    if ( l_e3sm_config ) then

      invrs_tau_zm = one_half * invrs_tau_zm

      do k = 1, nz
        do i = 1, ngrdcol
          invrs_tau_xp2_zm(i,k) = invrs_tau_bkgnd(i,k) + invrs_tau_sfc(i,k) + invrs_tau_shear(i,k) &
                            + C_invrs_tau_N2_xp2 * brunt_freq_pos (i,k)& ! 0
                            + C_invrs_tau_sfc * two &
                            * sqrt(em(i,k)) / ( gr(i)%zm(k) - sfc_elevation(i) + z_displace )  ! small
        end do
      end do

      invrs_tau_xp2_zm = min( max( sqrt( ( ddzt( nz, ngrdcol, gr, um)**2 + ddzt(nz, ngrdcol, gr, vm)**2 ) &
                        / max( 1.0e-7_core_rknd, brunt_vaisala_freq_sqd_smth ) ), &
                        0.3_core_rknd ), one ) * invrs_tau_xp2_zm

      invrs_tau_wpxp_zm = two * invrs_tau_zm &
                         + C_invrs_tau_N2_wpxp * brunt_freq_out_cloud

    else ! l_e3sm_config = false

      invrs_tau_xp2_zm =  0.1_core_rknd * invrs_tau_bkgnd + invrs_tau_sfc &
            + invrs_tau_shear + C_invrs_tau_N2_xp2 * brunt_freq_pos

      invrs_tau_xp2_zm = merge(0.003_core_rknd, invrs_tau_xp2_zm, &
            zt2zm(nz, ngrdcol, gr, ice_supersat_frac) <= 0.01_core_rknd &
            .and. invrs_tau_xp2_zm  >= 0.003_core_rknd)

      invrs_tau_wpxp_zm = invrs_tau_zm + C_invrs_tau_N2_wpxp * brunt_freq_out_cloud

    end if ! l_e3sm_config
~~~

**Reply by bstephens82** [reply]
Actually, I think I had that backwards, just FYI.  I think the taus code was updated to improve results in E3SM, and then those changes were merged into the CLUBB repo, but the changes degraded the CLUBB standalone runs.  So it's more like the flag was introduced to maintain backward-compatibility with the earlier CLUBB standalone results.

**Reply by adamrher** [reply]
thanks Ben. Since this only impacts the tau's, it guess it won't make a difference if ```l_diag_Lscale_from_tau=F```. Since Colin is recommending it, I'd vote for defaulting it to =T.

**Reply by zarzycki** [reply]
Yeah, agree with @adamrher's logic. Let's keep it true for now.

The other benefit is it keeps the configuration more apples<->apples with E3SM for model comparisons (i.e., the tuning coefficients should "directly" translate).

**Reply by bstephens82** [reply]
@zarzycki and @adamrher this may have already been clear to you guys, but something that clicked for me a couple days ago is that the C_invrs_tau* tunable parameters in CLUBB only matter if l_diag_Lscale_from_tau=T.  If that flag is false, these parameters aren't used.  So actually for the C_invrs_tau* parameters, I don't think we'd need to have a conditional in the namelist settings---we could just set them to be whatever we want them to be when the flag is T (e.g. Colin's parameters for now).

However, in a conversation with Vince on Friday, he said that the *intent* behind the C_invrs_tau* parameters was that, if they *are* used (i.e. if diag_Lscale=T), then C1, C2, C4, C6, C8, and C14 should all be set to 1.  The idea is that in the CLUBB equations, these parameters appear as numerators where the denominator is the associated tau, i.e. C1 appears in the wp2 equation as C1/tau, where tau is affected by the C_invrs_tau* parameters (if diag_Lscale=T).  So the idea is that if we're tuning the C_invrs_tau* parameters, then we can just set these "C" parameters to unity and avoid tuning them (although not *all* of the "C" parameters---C5, C7, and C11 for example are not divided by taus in the CLUBB equations).  So, at least in Vince's mind, ideally if diag_Lscale=T we would set the appropriate "C" parameters to 1 (here's where we could use a conditional in the namelist files), and then use appropriately tuned values for the C_invrs_tau* parameters.

Anyway this was Vince's thinking, but in order to set it up this way we might have to wait on some new tuning runs from Colin, since not all of the appropriate "C" parameters are set equal to 1 in his code currently.

**Reply by zarzycki** [reply]
@bstephens82 Those settings above are tuned with the correct "C" coefficients in my sims when diag_Lscale=T. See the block of code below. Some of the C6 terms are set to "2" -- I asked @vlarson about that and his response (Dec. 2021) was:

> The value of 2 is a historical artifact, and it is preserved in our code so that we can recover Zhun's E3SM tuning.
> 
> Someday we'll get rid of it, but the key point is not the particular value of C6, but that if you're tuning, you should tune the C_invrs_... parameters and not C6, C1, etc., because the latter parameters are redundant, and we don't want the space of tuning parameters to be too large.

These do match the settings Zhun Guo used when first working with the taus code in E3SM. A file containing these tunings is accessible in the [CLUBB repo](https://github.com/larson-group/clubb_release/blob/master/input/tunable_parameters_e3sm/tunable_parameters.in).

This is what I was referring to re: the conditional tuning. That is, C1 will have a different value if diag_Lscale=T (1.0) and diag_Lscale=F (whatever the default C1 is). Sorry for the confusion.

Required "preconditions" for running diag_Lscale=T (in addition to the c_invrs_tau terms).

```
clubb_C1          = 1.000000 ! Low Skewness in C1 Skewness Function. Units [-]
clubb_C1b         = 1.000000 ! High Skewness in C1 Skewness Function. Units [-]
!clubb_C1c         = 1.000000 ! Degree of Slope of C1 Skewness Function. Units [-]
clubb_C2rt        = 1.000000 ! C2 constant associated with rtp2 dissipation term. Units [-]
clubb_C2thl       = 1.000000 ! C2 constant associated with thlp2 dissipation term. Units [-]
clubb_C2rtthl     = 1.000000 ! C2 constant associated with rtpthlp dissipation term. Units [-]
clubb_C6rt        = 2.000000 ! Low Skewness in C6rt Skewness Function. Units [-]
clubb_C6rtb       = 2.000000 ! High Skewness in C6rt Skewness Function. Units [-]
clubb_C6rtc       = 1.000000 ! Degree of Slope of C6rt Skewness Function. Units [-]
clubb_C6thl       = 2.000000 ! Low Skewness in C6thl Skewness Function. Units [-]
clubb_C6thlb      = 2.000000 ! High Skewness in C6thl Skewness Function. Units [-]
clubb_C6thlc      = 1.000000 ! Degree of Slope of C6thl Skewness Function. Units [-]
clubb_C14         = 1.000000 ! Constant applied to up2 and vp2. Units [-]
```

**Reply by bstephens82** [reply]
@zarzycki ok, cool, thanks, I wasn't sure if what Vince was saying to me was clear to everyone so I thought I should document it here. Sounds like we are on the right track.

**Reply by zarzycki** [reply]
Yes, it took me a little while to figure out exactly what was going on as well (in Zhun's paper, he references "removing" the coefficients, but in reality it's just setting them to unity so the code is backwards compatible without having multiple if/else blocks).

I think this is why it would be a good idea to have a conditional set of tuning for diag_Lscale=T. Otherwise, unless we want to put a "trap" in clubb_intr, it's too easy to change c_invrs_tau terms without enforcing the other C coefficients to unity (the code will happily run, though, because you are essentially scaling the tau profile by the C terms .ne. 1!).

**Reply by adamrher** [reply]
Things are becoming clearer to me; it is the pr1 and dp1 coefficients (C1, C2, C4, C6, C8, and C14) that should be set to 1 when diag_Lscale=T. @bstephens82 so the idea for having the namelists conditionals would trigger both (1) setting al the pr1 and dp1 terms to 1., and (2) adjust a separate group of namelists (not related to dp1 or pr1 terms, but clubb_gamma, for example), since the climate is degraded by using Colin's tau values but no changes to group (2). Although, I suppose the figure you showed w/ the excessive SWCF is also a run where the dp1 and pr1 terms were not set to 1, and so that makes it a bit of an outlier of a run.

I do see some differences in the list of Guo's namelists Colin pointed to [here](https://github.com/larson-group/clubb_release/blob/master/input/tunable_parameters_e3sm/tunable_parameters.in). Not only as Colin said, are the C6 variables set to 2 ... the skewness dependent part C6c is set to 1. Maybe this is on purpose, to still allow the dp1 and pr1 terms to have skewness dependence since the invrs tau vars don't have this functionality? 

The only other mystery is why Guo's C8/C8b is set to 0.5/0.02. This refers to the pr1 term for w'3. Not sure why these are not-equal to 1.

**Reply by bstephens82** [reply]
I also notice that Guo's C4=2, but maybe this is another "historical artifact", not sure.

**Reply by vlarson** [reply]
> I do see some differences in the list of Guo's namelists Colin pointed to [here](https://github.com/larson-group/clubb_release/blob/master/input/tunable_parameters_e3sm/tunable_parameters.in). Not only as Colin said, are the C6 variables set to 2 ... the skewness dependent part C6c is set to 1. Maybe this is on purpose, to still allow the dp1 and pr1 terms to have skewness dependence since the invrs tau vars don't have this functionality?

Sorry for the confusion.  If C6=C6b, then the value of C6c should have no impact on the results.  The "c" coefficients merely set the transition width between the "" and "b" values.
 
> The only other mystery is why Guo's C8/C8b is set to 0.5/0.02. This refers to the pr1 term for w'3. Not sure why these are not-equal to 1.

We wanted to leave a tuning parameter to help determine the ratio of damping on wp2 and wp3.   We didn't have time to figure out a convenient way to do that using the new taus.

**Reply by vlarson** [reply]
> If I'm interpreting Coln's earlier post correctly (likely I'm not), I think we should have conditionals in the namelist_defaults that recovers our current namelists if `l_diag_Lscale_from_tau=F` and otherwise will use another complete set of clubb namelists if `l_diag_Lscale_from_tau=T`. As a start we can plug in Colin's values above. But at least this will give something approaching backward compatibility if we run with ldiag=F.
> 
> Does everyone agree? If so, I'll add this feedback to Kate's PR.

Based on this discussion, I'm thinking that maybe UWM should place a trap inside CLUBB that will stop the run with an informative error if `l_diag_Lscale_from_tau=T` and C1 .ne. 1, C2 .ne. 1, etc.

**Reply by vlarson** [reply]
> I also notice that Guo's C4=2, but maybe this is another "historical artifact", not sure.

Good point.  Ultimately we should make C4=1 too.

**Reply by adamrher** [reply]
Hi All,

I made a branch of Kate's branch with a first crack at a namelist_defaults. @bstephens82 could you look through this and see what I'm missing: 

https://github.com/adamrher/CAM/blob/5759a2c7c5b0f5308d420c8dafdb3af37458122d/bld/namelist_files/namelist_defaults_cam.xml#L1908-L1942

I don't see the e3sm flag yet in Kate's branch. So that has to be added in. I think everything conditioned on ```clubb_l_diag_Lscale_from_tau``` is set to Colin's preferred values.

**Reply by bstephens82** [reply]
@adamrher in personal communication with @zarzycki he has indicated he's using the following values for a few of the C_invrs_tau parameters:
~~~
 C_invrs_tau_N2_wp2           =   0.100000000000000
 C_invrs_tau_N2_xp2           =   5.000000000000000E-002
 C_invrs_tau_N2_wpxp          =   0.000000000000000E+000
 C_invrs_tau_N2_clear_wp3     =    1.00000000000000
~~~
which differ from the values in your namelist.

Other things:  
Colin seems to be using C4=5.2 (same as the diag_Lscale=F case according to the CAM namelist), so you can probably just drop that line.
Also your clubb_c11 for diag_Lscale=T is formatted incorrectly with two decimal points.

Those are the only things I've noticed for now.

**Reply by adamrher** [reply]
thanks, done. I kept a duplicate C4=5.2 for Lscale=T, even though it's redundant (I think we should have all dp1 and pr1 terms duplicated for Lscale=T, even if they're the same as =F).

Im looking into the e3sm namelist now.

**Reply by adamrher** [reply]
@bstephens82 do you still have any of the 2 runs at the top of this thread? I'd like to cross reference the atm_in file with the defaults I'm working on (atm_in is usually in scratch/username/case/run/atm_in/)

**Reply by bstephens82** [reply]
I don't have the output anymore for the first two runs mentioned in the thread, but the atm_in (prognostic momentum flux T and F) files from the [case]/CaseDocs directories are here:
[atm_in.pmfF.txt](https://github.com/NCAR/amwg_dev/files/8413140/atm_in.pmfF.txt)
[atm_in.pmfT.txt](https://github.com/NCAR/amwg_dev/files/8413141/atm_in.pmfT.txt)
